File: source/api/lines/line.model.js
//
// ─── LINE MODEL ─────────────────────────────────────────────────────────────────
//

const mongoose = require('mongoose');

const { Schema } = mongoose; 

const LineSchema = new Schema({
  line: { type: String, required: true },
  characterId: { type: Schema.Types.ObjectId, ref: 'Character' }, 
  episodeId: { type: Schema.Types.ObjectId, ref: 'Episode' }, 
});

LineSchema.pre('find', function (next) {
  this.populate('lines')
  next()
})

LineSchema.pre('findOne', function (next) {
  this.populate('lines')
  next()
});

const Line = mongoose.model('Line', LineSchema); 

module.exports = {
  Line,
};

File: models/line.model.go
package models

type Line struct {
	ID 					bson.ObjectId `json:"_id,omitempty" bson:"_id,omitempty"`
	Line 				string 				`required:"true"`
	CharacterId	bson.ObjectId	`json:"characterId,omitempty" bson:"characterId,omitempty"`
	EpisodeId		bson.ObjectId	`json:"episodeId,omitempty" bson:"episodeId,omitempty"`
}